src/core/library.c

```diff
@@ -1112,6 +1112,30 @@ QuicLibrarySetGlobalParam(
         Status = QUIC_STATUS_SUCCESS;
         break;
 
+    case QUIC_PARAM_GLOBAL_STATELESS_RESETKEY:
+        if (BufferLength != QUIC_STATELESS_RESETKEY_LENGTH * sizeof (uint8_t)) {
+            Status = QUIC_STATUS_INVALID_PARAMETER;
+            break;
+        }
+
+        for (uint16_t i = 0; i < MsQuicLib.ProcessorCount; ++i) {
+            QUIC_LIBRARY_PP* PerProc = &MsQuicLib.PerProc[i];
+            CxPlatLockAcquire(&PerProc->ResetTokenLock);
+            CxPlatHashFree(PerProc->ResetTokenHash);
+            Status =
+                CxPlatHashCreate(
+                    CXPLAT_HASH_SHA256,
+                    (uint8_t*)Buffer,
+                    QUIC_STATELESS_RESETKEY_LENGTH * sizeof(uint8_t),
+                    &PerProc->ResetTokenHash);
+            if (QUIC_FAILED(Status)) {
+                return Status;
```

Comment:
Need to set the pointer to NULL on failure, so it's not tried to free again later. 
```suggestion
                PerProc->ResetTokenHash = NULL;
                return Status;
```
But actually, the more I think about it, if we fail and leave one of these NULL, we're in bad shape. We should probably:

1. create the new hash first
2. if it succeeds:
  a. lock
  b. swap the pointers
  c. unlock
  d. free the old hash

Reply:
How about we do the same way as random reset key function, we call MsQuicLibraryFreePartitions() to reset PerProc->ResetTokenHash at failure case?

Reply:
Make sense. One more thing, should we "free the old hash" before "swap the pointers"?